src/extra/components/timepix.py

```diff
@@ -124,26 +130,65 @@ def _prepare_pasha(parallel):
         """Try to find detector source."""
 
-        detectors = defaultdict(list)
+        if isinstance(prefix_or_source, tuple) and len(prefix_or_source) == 2:
+            # Explicit tuple of source names.
 
-        for source in data.instrument_sources:
-            m = cls._instrument_re.match(source)
-            if m is not None and (not domain or m[1] == domain):
-                detectors[m[1]].append(source)
+            def _find_sources(data, source):
+                sd = data[source]
+                domain = source.partition('/')[0]
 
-        if len(detectors) > 1:
-            raise ValueError('multiple detectors found, please pass one '
-                             'explicitly:\n' + ', '.join(sorted(detectors)))
-        elif detectors:
-            return next(iter(detectors.items()))
+                if sd.is_control and f'{source}:daqOutput.chip0' in data.all_sources:
+                    return domain, f'{source}:daqOutput.chip0'
+                elif sd.is_instrument:
+                    return domain, source
 
-        if domain:
-            raise ValueError(f'no sources found for detector {domain}')
-        else:
-            raise ValueError('no detector found, please pass one explicitly')
+            raw_source, centroided_source = prefix_or_source
+
+            if not raw_source and not centroided_source:
+                raise ValueError('tuple of source names may not be all empty')
+
+            found_sources = []
+
+            if raw_source:
+                domain, source = _find_sources(data, raw_source)
+                found_sources.append(source)
+
+            if centroided_source:
+                domain, source = _find_sources(data, centroided_source)
+                found_sources.append(source)
+
+            return domain, found_sources
+
+        elif isinstance(prefix_or_source, str):
+            # Detector domain.
+
+            detectors = defaultdict(list)
+
+            for source in data.instrument_sources:
+                m = cls._instrument_re.match(source)
+                if m is not None and m[1].startswith(prefix_or_source):
+                    detectors[m[1]].append(source)
+
+            if len(detectors) > 1:
+                raise ValueError('multiple detector domains found, please '
+                                 'pass one explicitly:\n' +
+                                 ', '.join(sorted(detectors)))
+            elif detectors:
+                return next(iter(detectors.items()))
+
+            if prefix_or_source:
+                raise ValueError(f'no detector sources found for '
+                                 f'{prefix_or_source}, please pass explicit '
+                                 f'source name(s)')
+            else:
+                raise ValueError('no detector detected, please narrow the '
+                                 'search with the `detector` argument')
+
+        raise ValueError('detector may be a string, tuple of two strings or '
```

Comment:
```suggestion
        raise TypeError('detector may be a string, tuple of two strings or '
```

Reply:
Fair point, I tend to use `ValueError` too much.